Yoast/YoastSEO.js
# Add `spec` to code style checks

## Summary

This PR can be summarized in the following changelog entry:

* _Not applicable_

## Relevant technical choices:

* Added a separate eslint configuration file, to make sure the specific test requirements don't leak into the source files.

## Test instructions

This PR can be tested by following these steps:

* Run `grunt check` and see all tests pass
* Run `grunt eslint:tests` to only run the specs tests
* Run `yarn test` to make sure the tests are unaffected by the changes

Fixes #1452
